### ISSUE TITLE
Shader not working for Android 10?

### ISSUE THREAD

**Issue by Quillraven** [Quillraven/Dark-Matter#6]
@czyzby @tommyettinger: Seems like there is a problem with Android 10 that fetchUniformLocation is not working. Do you have any experience with that or how to solve it?

Otherwise, I will google a little bit over the weekend to find that out.

I tested it with an emulated device running Android 10 and the game crashes with below stacktrace when switching to a custom shader. Could also be related that the custom shader is not loaded properly and for some reason it does not get logged correctly.

Running fine on my Galaxy S8 though. Not running on Galaxy S10 of a friend ;)

```
E/AndroidRuntime: FATAL EXCEPTION: GLThread 551
    Process: com.github.quillraven.darkmatter, PID: 7275
    java.lang.IllegalArgumentException: no uniform with name 'u_projTrans' in shader
        at com.badlogic.gdx.graphics.glutils.ShaderProgram.fetchUniformLocation(ShaderProgram.java:308)
        at com.badlogic.gdx.graphics.glutils.ShaderProgram.fetchUniformLocation(ShaderProgram.java:298)
        at com.badlogic.gdx.graphics.glutils.ShaderProgram.setUniformMatrix(ShaderProgram.java:528)
        at com.badlogic.gdx.graphics.glutils.ShaderProgram.setUniformMatrix(ShaderProgram.java:519)
        at com.badlogic.gdx.graphics.g2d.SpriteBatch.setupMatrices(SpriteBatch.java:1061)
        at com.badlogic.gdx.graphics.g2d.SpriteBatch.setShader(SpriteBatch.java:1091)
        at com.github.quillraven.darkmatter.ecs.system.RenderSystem.renderEntityOutlines(RenderSystem.kt:100)
        at com.github.quillraven.darkmatter.ecs.system.RenderSystem.update(RenderSystem.kt:95)
        at com.badlogic.ashley.core.Engine.update(Engine.java:202)
        at com.github.quillraven.darkmatter.screen.MenuScreen.render(MenuScreen.kt:81)
        at ktx.app.KtxGame.render(game.kt:53)
        at com.github.quillraven.darkmatter.Game.render(Game.kt:131)
        at com.badlogic.gdx.backends.android.AndroidGraphics.onDrawFrame(AndroidGraphics.java:494)
        at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1573)
        at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1272)

```

**Comment by Quillraven** [comment]
guess I found it - I did something wrong in the fragment shader :(

```
E/AssetStorage: ShaderProgram shader/default.vert/outline.frag failed to compile:
    Fragment shader:
    ERROR: 0:1: '' : No precision specified for (float) 
    ERROR: 0:2: '' : No precision specified for (float) 
    ERROR: 0:5: '' : No precision specified for (float) 
    ERROR: 0:6: '' : No precision specified for (float) 
    ERROR: 0:14: '' : No precision specified for (float) 
```

**Comment by tommyettinger** [comment]
This kind of shader bug is incredibly easy to encounter, but I hadn't yet seen it happen on a Galaxy S10 and not a Galaxy S8. For a quick response from libGDX's large community, you can try the Discord, https://discord.gg/nVT3Xgh , which has a fairly world-wide set of people who have probably hit the same bug. If you're still hitting shader issues, how can I help?

**Comment by Quillraven** [comment]
thank you tommyettinger! initially I thought it was maybe a gradle android setup issue that is why I included you but it wasn't. 

it was my own fault and I saw that in the spritebatch class, where the defaultshaders are created, there is a block to define a precision.

I added this block now to my own fragment shader and waiting for the feedback of the testers, if it solved the issue.

Just wondering why it worked in my S8 but it didn't work on S8+ and S10 of my testers. But hopefully we soon will find out if this precision block solves it  ;)

Thank you anyway!
